pkg/image/api/v1/types.go

```diff
@@ -42,6 +44,73 @@ type ImageLayer struct {
 	Size int64 `json:"size"`
 }
 
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob.
+	Type string `json:"type"`
+	// An opaque binary string which is an image's signature.
+	Content []byte `json:"content"`
+	// Conditions represent the latest available observations of a signature's current state.
+	Conditions []SignatureCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	// Following metadata fields will be set by server if the signature content is successfully parsed and
+	// the information available.
+
+	// A human readable string representing image's identity. It could be a product name and version, or a
+	// image pull spec.
+	ImageIdentity string `json:"imageIdentity,omitempty"`
+	// Contains claims from the signature.
+	SignedClaims map[string]string `json:"signedClaims,omitempty"`
+	// If specified, it is the time of signature's creation.
+	Created *unversioned.Time `json:"created,omitempty"`
+	// If specified, it holds information about an issuer of signing certificate or key (a person or entity
+	// who signed the signing certificate or key).
+	IssuedBy SignatureIssuer `json:"issuedBy,omitempty"`
+	// If specified, it holds information about a subject of signing certificate or key (a person or entity
+	// who signed the image).
+	IssuedTo SignatureSubject `json:"issuedTo,omitempty"`
+}
+
+/// SignatureConditionType is a type of image signature condition.
+type SignatureConditionType string
+
+// SignatureCondition describes an image signature condition of particular kind at particular probe time.
+type SignatureCondition struct {
+	// Type of job condition, Complete or Failed.
+	Type SignatureConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status kapi.ConditionStatus `json:"status"`
+	// Last time the condition was checked.
+	LastProbeTime unversioned.Time `json:"lastProbeTime,omitempty"`
```

Comment:
> Why do we need Probe and Transition time?

Some conditions can change over time:  `Trusted`, `Revoked`, `Expired`  come to mind.  Knowing when it was last checked and last changed matters.